main.c

```diff
@@ -9,128 +10,413 @@
+                        scanf("%i", &selection);
+                        if (selection == 1) {
+                            printf("1. Inactive\t2. Active\t3. Delinquent\t4. Deleted\n");
+                            printf("Input user's title: ");
+                            scanf("%i", &status);
+                            printf("You have entered: %i\n", status);
+                        }
+                        User* filterBy = createUser(user_id, full_name, login_name, NULL, email, title, status);
+                        printUser(filterBy);
+                        User* filtered_users = filterUsers(filterBy, user_list_head);
+                        if (filtered_users) {
+                            printf("%i user(s) found\n", countUsers(filtered_users));
+                            printUser(filtered_users);
+                        }
+                        else    printf("No user found!\n");
+                        freeUserList(filtered_users);
+                        break;
+                    case 5:
+                        printf("Enter new user's information\n");
+                        int new_user_id = 0;
+                        char* new_full_name = malloc(100);
+                        char* new_login_name = malloc(50);
+                        char* new_email = malloc(100);
+                        char* new_password = malloc(50);
+                        enum credential new_title = 0;
+                        printf("Enter user ID: ");
+                        scanf("%i", &new_user_id);
+                        printf("Enter user's full name: ");
+                        scanf(" %[^\n]", new_full_name);
+                        printf("Enter user's login name: ");
+                        scanf(" %[^\n]", new_login_name);
+                        printf("Enter user's email: ");
+                        scanf(" %[^\n]", new_email);
+                        printf("Enter user's password: ");
+                        scanf(" %[^\n]", new_password);
+                        printf("1. Author\t2. Customer\t3. Librarian\n"
+                               "Enter user's title: ");
+                        scanf("%i", &new_title);
+                        User* new_user = createUser(new_user_id, new_full_name, new_login_name, new_password, new_email, new_title, active_user);
+                        insertUser(new_user, &user_list_head);
+                        printf("New user added!\n");
+                        printUser(new_user);
+                        break;
+                    case 6:
+                        printf("Enter user ID to be deleted: ");
+                        int id = 0;
+                        scanf("%i", &id);
+                        int deleted_user = deleteUser(id);
+                        if (deleted_user > 0)  printf("%i user deleted!\n", deleted_user);
+                        else    printf("No user found!\n");
+                        break;
+                }
```

Comment:
If user is author, his books should be inactive as well 
if user is customer, all his transactions should be inactive and book copies count should update